app/extractors/BeneficiaryExtractor.scala

```diff
@@ -117,4 +181,42 @@ trait BeneficiaryExtractor[T <: Beneficiary] {
     }
   }
 
+  def extractIdentification(identification: Option[IndividualIdentification],
+                            address: Option[Address],
+                            answers: UserAnswers) : Try[UserAnswers] = {
+    if (answers.isTaxable) {
+      identification match {
+        case Some(NationalInsuranceNumber(nino)) =>
+          answers.set(nationalInsuranceNumberYesNoPage, true)
+            .flatMap(_.set(nationalInsuranceNumberPage, nino))
+        case Some(p: Passport) =>
+          answers.set(nationalInsuranceNumberYesNoPage, false)
+            .flatMap(_.set(passportOrIdCardDetailsYesNoPage, true))
+            .flatMap(_.set(passportOrIdCardDetailsPage, p.asCombined))
+        case Some(id: IdCard) =>
+          answers.set(nationalInsuranceNumberYesNoPage, false)
+            .flatMap(_.set(passportOrIdCardDetailsYesNoPage, true))
+            .flatMap(_.set(passportOrIdCardDetailsPage, id.asCombined))
+        case Some(combined: CombinedPassportOrIdCard) =>
+          answers.set(nationalInsuranceNumberYesNoPage, false)
+            .flatMap(_.set(passportOrIdCardDetailsYesNoPage, true))
+            .flatMap(_.set(passportOrIdCardDetailsPage, combined))
+        case _ =>
+          answers.set(nationalInsuranceNumberYesNoPage, false)
+            .flatMap(answers => extractPassportOrIdCardDetailsYesNo(address.isDefined, answers))
+      }
+    } else {
+      Success(answers)
+    }
+  }
+
+  def extractPassportOrIdCardDetailsYesNo(hasAddress: Boolean, answers: UserAnswers): Try[UserAnswers] = {
+    if (answers.isTaxable && hasAddress) {
+      answers.set(passportOrIdCardDetailsYesNoPage, false)
+    } else {
+      Success(answers)
+    }
+  }
```

Comment:
Likewise I would keep this in the `IndividualBeneficiaryExtractor`

Reply:
Updated